pulser-core/pulser/sequence/sequence.py

```diff
@@ -364,6 +365,129 @@ def config_slm_mask(self, qubits: Iterable[QubitId]) -> None:
+            for n_d_ch_id, n_d_ch_obj in new_device.channels.items():
+                # Find the corresponding channel on the new device
+                # We verify the channel class then
+                # check whether the addressing Global or local
+                basis_match = o_d_ch_obj.basis == n_d_ch_obj.basis
+                addressing_match = (
+                    o_d_ch_obj.addressing == n_d_ch_obj.addressing
+                )
+                if not (basis_match and addressing_match):
+                    channel_match[o_d_ch_name] = None
+                    ch_type_er_mess = f"No match for channel {o_d_ch_name}."
+                    continue
+                if not strict:
+                    channel_match[o_d_ch_name] = n_d_ch_id
+                    break
+                ch_samples = sample_seq.channel_samples[o_d_ch_name]
+                ch_sample_phase = ch_samples.phase
+                # Find if there is phase change between pulses or not
+                phase_is_constant = True
+                if ch_sample_phase.size != 0:
+                    phase_is_constant = bool(np.all(
+                        ch_sample_phase == ch_sample_phase[0]
+                    ))
+                # Phase_jump_time and clock_period check
+                phase_jump_time_check = (
+                    o_d_ch_obj.phase_jump_time == n_d_ch_obj.phase_jump_time
+                )
+                clock_period_check = (
+                    o_d_ch_obj.clock_period % n_d_ch_obj.clock_period == 0
+                )
+                if clock_period_check and (
+                    phase_is_constant or phase_jump_time_check
+                ):
+                    channel_match[o_d_ch_name] = n_d_ch_id
+                    alert_phase_jump = not phase_jump_time_check
+                    break
+                else:
+                    channel_match[o_d_ch_name] = None
+                    strict_error_message = (
+                        "No channel with phase_jump_time"
+                        + " & clock_period match."
+                    )
+
+        if None in channel_match.values():
+            if strict_error_message:
+                raise ValueError(strict_error_message)
+            else:
+                raise TypeError(ch_type_er_mess)
+        # Initialize the new sequence
+        new_seq = Sequence(self.register, new_device)
+
+        for call in self._calls[1:]:
+            if not (call.name == "declare_channel"):
+                getattr(new_seq, call.name)(*call.args, **call.kwargs)
+                continue
+            # Switch the old id with the correct id
+            sw_channel_args = list(call.args)
+            sw_channel_kw_args = call.kwargs.copy()
+            if "name" in sw_channel_kw_args:
+                sw_channel_kw_args["channel_id"] = channel_match[
+                    sw_channel_kw_args["name"]
+                ]
+            elif "channel_id" in sw_channel_kw_args:
```

Comment:
First add a comment saying why these cases are not touched. 
Then add these lines:
```suggestion
            if "name" in sw_channel_kw_args:    # pragma: no cover
                sw_channel_kw_args["channel_id"] = channel_match[
                    sw_channel_kw_args["name"]
                ]
            elif "channel_id" in sw_channel_kw_args:    # pragam: no cover
```